src/lib/selectors/youtube/main.py

```diff
@@ -7,14 +7,15 @@
 
 
 class YoutubeSelector(Selector):
-    def index(self, config):
+    def index_files(self, config):
         if not os.path.exists(self.SELECT_MAP):
             df, logs = selector_run(config, self.FOLDER)
```

Comment:
The `selector_run` method should be refactored so that it doesn't manually keep track of logs, but instead calls `self.logger` in the appropriate places. `self.logger` can be handed down as an argument on `selector_run`-- or better, the entire Youtube selector should be refactored into this 'main.py' file.

Reply:
Refactored entire youtube selector into main.py